honeycomb-core/src/cells/attributes.rs

```diff
@@ -0,0 +1,107 @@
+//! Generic attributes implementation
+//!
+//! This module contains all code used to handle attribute genericity in the context of mesh
+//! representation through combinatorial maps embedded data.
+
+// ------ IMPORTS
+
+use crate::OrbitPolicy;
+
+// ------ CONTENT
+
+/// Generic attribute trait for logical behavior
+///
+/// This trait can be implemented for a given attribute in order to define the behavior to
+/// follow when (un)sewing operations result in an update of the attribute.
+///
+/// # Example
+///
+/// For an intensive property of a system (e.g. a temperature), a dummy implementation would look
+/// like this:
+///
+/// ```rust
+/// use honeycomb_core::AttributeLogic;
+///
+/// #[derive(Copy, Clone)]
+/// pub struct Temperature {
+///     pub val: f32
+/// }
+///
+/// impl AttributeLogic for Temperature {
+///     fn merge(lhs: Self, rhs: Self) -> Self {
+///         Temperature { val: (lhs.val + rhs.val) / 2.0 }
+///     }
+///
+///     fn split(lhs: Self) -> (Self, Self) {
+///         (lhs, lhs)
+///     }
+///
+///     fn merge_undefined(lhs: Option<Self>) -> Self {
+///         lhs.unwrap_or(Temperature { val: 0.0 })
+///     }
+/// }
+/// ```
+pub trait AttributeLogic: Sized {
+    /// Merging routine, i.e. how to obtain the new attribute value from the two existing ones.
+    fn merge(lhs: Self, rhs: Self) -> Self;
+
+    /// Splitting routine, i.e. how to obtain the two attributes from a single one.
+    fn split(lhs: Self) -> (Self, Self);
+
+    /// Fallback merging routine, i.e. how to obtain the new attribute value from potentially
+    /// undefined instances.
+    ///
+    /// The default implementation may panic if no attribute can be used to create a value. The
+    /// reason for that is as follows:
+    ///
+    /// This trait and its methods were designed with the (un)sewing operation in mind. Their
+    /// purpose is to simplify the code needed to propagate updates of attributes affected by the
+    /// (un)sewing operation. Considering this context, as well as the definition of (un)linking
+    /// operations, this panic seems reasonable: If the darts you are sewing have totally undefined
+    /// attributes, you should most likely be linking them instead of sewing.
+    fn merge_undefined(lhs: Option<Self>) -> Self {
+        lhs.unwrap()
+    }
+}
```

Comment:
Comparing to `bind_to`, these are functions and not methods, is there any reason to treat them differently?

Also it is not very clear what `merge_undefined` can do with 0 or 1 attribute to *merge*.

Reply:
I'm actually not sure why I made `binds_to` a method instead of a function. Both traits are meant to describe the **behavior of the attribute** (i.e. a type, structure, enum...), **not the behavior of a given instance of the attribute**. 

Defining these functions as methods also introduce some asymmetry to arguments which I don't think is coherent with what I described above.